silkworm/rpc/http/connection.cpp

```diff
@@ -190,10 +192,11 @@ Task<void> Connection::handle_actual_request(const boost::beast::http::request<b
     vary_ = req[boost::beast::http::field::vary];
     origin_ = req[boost::beast::http::field::origin];
     method_ = req.method();
+    auto encoding = req[boost::beast::http::field::accept_encoding];
 
     auto rsp_content = co_await request_handler_.handle(req.body());
     if (rsp_content) {
-        co_await do_write(rsp_content->append("\n"), boost::beast::http::status::ok);
+        co_await do_write(rsp_content->append("\n"), boost::beast::http::status::ok, !encoding.empty());
```

Comment:
The value of the `Accept-Encoding` field should be passed to the `do_write` method instead of a `bool`. This will allow for checking within `do_write` which kind of content compression has been requested and accept/reject it depending on the supported compression on our side